### ISSUE TITLE
TSC Subcommittee Meeting (15 Apr 2021): Partnership Program

### ISSUE THREAD

**Issue by mickmcgrath13** [StackStorm/discussions#73]
April 2021 @StackStorm/tsc 1 hour meeting will take place on Thursday, 15th Apr 2021, 12:00 PM US Pacific.
This event has a video call.
Join: https://meet.google.com/tdq-xvwo-mnu
(US) +1 650-735-3248 PIN: 128167492#
View more phone numbers: https://tel.meet/tdq-xvwo-mnu?pin=2872811864098&hs=7

# Purpose

To decide on a partnership program.  Many of these discussion topics will have been answered or discussed in other issues (like [this one](https://github.com/StackStorm/discussions/issues/72)).  This issue is meant to aggregate them and start a subcommittee to take them forward.

# Goals
- To align incentives for the StackStorm project as well as partners
    - Give and take - two-way significant value
    - What does ST2 provide to partners?  What does it explicitly not provide?
    - What do partners provide to ST2?  What do they explicitly not provide?
 
# Deadline
Final draft by next ST2 TSC meeting

# Governance
Anyone can join and contribute ideas.

Led initially by 4 partners:
- ~Amanda McGuinness (@amanda11)~ Edwin Lewzey (@elewzey)
- JP Bourget (@punkrokk)
- Ravi Rao (@raviorch)
- Mick McGrath (@mickmcgrath13)

## Rules
- What is okay?
    - What is in the scope of access to a partner?
- What is not okay?
    - What is not in the scope of access to a partner?

## FAQ
Should we create/host an FAQ for a 'soft landing' of governance / conflict of interest?

## Disputes
In general, consensus for involved parties should be sufficient.
If a dispute arises, how are they resolved?  This could be: 
- Raise to TSC for a vote?
- or Dedicated arbitration guidance/framework for the subcommittee

# Mechanics
- use [existing issue](https://github.com/StackStorm/discussions/issues/72) or a new one?
- new issue for subcommittee?
- dedicated slack channel?
- Where will the draft be created (github issue, Google Drive doc, etc)?

**Comment by raviorch** [comment]
Thanks Mick, for creating the issue. Will post here the recommendation that I sent out to the other thread.

**Comment by mickmcgrath13** [comment]
This one:
https://github.com/StackStorm/discussions/issues/72#issuecomment-814239951
?

**Comment by mickmcgrath13** [comment]
We should have verbiage to mitigate legal risk from incoming partners in case of possible "bad actors".  Maybe something simple along the lines of:
> I, as a partner, accept to not hold StackStorm accountable [...]

**Comment by raviorch** [comment]
Exactly we need incorporate this type of language. I will work with our legal counsel to get this sorted out if necessary.

**Comment by amanda11** [comment]
For Ammeon representation I'll be delegating to @elewzey who was also on the call. (I lost my mic half way through...)

**Comment by mickmcgrath13** [comment]
![image](https://user-images.githubusercontent.com/1030325/113754830-408b3f00-96d5-11eb-839b-7ebc414673ba.png)

**Comment by arm4b** [comment]
> We should have verbiage to mitigate legal risk from incoming partners in case of possible "bad actors". Maybe something simple along the lines of:
>>     I, as a partner, accept to not hold StackStorm accountable [...]

Here is what LF legal proposed and so it was drafted on the https://stackstorm.com/partners-program/ (beta):

>  In your [partnership email] proposal, please include the following statement:
> > On behalf of PARTNER NAME ("Participant"), I agree that Participant will comply with the StackStorm Partners Program requirements published at https://stackstorm.com/partners-program/ and Linux Foundation Policies published at https://lfprojects.org/policies/, as updated from time to time, during our participation in the Partners Program.

What would be the language to add to that statement? cc @raviorch

**Comment by raviorch** [comment]
Eugene,

We definitely would like to add verbiage like you mentioned below. In 
the meantime, you mentioned on the TSC call that you had researched 
other open source project partnership charters. Can you please provide 
examples of the same so the subcommittee has a starting point instead of 
reinventing the wheel. This would really accelerate our progress as 
other open source projects may have solved most of the issues we are 
seeing here. Please send this out asap as I will use that to put a draft 
out before the weekend if that is available.

Ravi

------ Original Message ------
From: "Eugen Cusmaunsa" ***@***.***>
To: "StackStorm/discussions" ***@***.***>
Cc: "raviorch" ***@***.***>; "Mention" 
***@***.***>
Sent: 4/6/2021 10:43:41 AM
Subject: Re: [StackStorm/discussions] Partnership Program Subcommittee 
(#73)

**Comment by raviorch** [comment]
Mick, JP, Edwin, please let me know what time is the best for us to meet. I have asked Eugene to provide some examples, in the meantime here are some examples I found: 

https://www.mongodb.com/partners/partner-program
https://partner.suse.com/English/
[SUSE-Application-Process-Final.pdf](https://github.com/StackStorm/discussions/files/6267065/SUSE-Application-Process-Final.pdf)
https://partners.gitlab.com/English/
There a bunch more, but as it clearly states in the GitLab Channel Partner page, https://about.gitlab.com/handbook/resellers/ 
It talks about what the community provides to the partner first before getting into what the partner should provide, this is the type of language that we should use.

**Comment by arm4b** [comment]
@raviorch These are good examples. However, you're referring to the projects with the Open Core + Enterprise model. This is what Extreme Networks was before with the EWC. I'd imagine EXTR would absolutely need a complex multi-tier Partners Program with the Legal Agreements, etc, etc as it's a B2B partnership.

This is a very different model compared to how StackStorm 100% Open Source operates under the neutral non-profit LF governance with different values and system behind.

Here is an example for the Debian Open Source Partner program that was a prototype for the https://stackstorm.com/partners/ and https://stackstorm.com/partners-program/ drafts.

**https://www.debian.org/partners/**
**https://www.debian.org/partners/2021/partners**

In short, they want to highlight orgs by providing recognition (listing on the page) as "official partner" for those significantly helping the project. StackStorm can go further and provide co-marketing opportunities (examples at [RFC: StackStorm Partners, Code of Conduct and Economy #51](https://github.com/StackStorm/discussions/issues/51). 

Similar to Debian, do you think we would be able to focus on simplicity and maintenance-free keeping in mind the Open Source community interests?

**Comment by mickmcgrath13** [comment]
## Meeting
@raviorch  I can meet for either of the following
- Wednesday, April 14 at 3 PM Eastern
- Thursday, April 15 at 3 PM Eastern

## Benefits
@raviorch  All of the links you posted (mongo, suse, gitlab) clearly display benefits of becoming a partner (to @punkrokk's point), and I think we should outline similar benefits.  @armab's [proposal](https://github.com/StackStorm/discussions/issues/72) does have a section for it already. It mostly speaks to 2 things:
- Redirect community inquires to the partners page
- Participate in shared marketing (blogs, etc)

Also, I like the table presentation in the [mongo partners page](https://www.mongodb.com/partners/partner-program):
![image](https://user-images.githubusercontent.com/1030325/113865417-ac22ea00-9771-11eb-87c5-38bec1e823fe.png)

GitLab has a similar one:
![image](https://user-images.githubusercontent.com/1030325/113866233-b09bd280-9772-11eb-95b4-71a8e5dfd3ed.png)

### Possible Tasks
- I think for the benefits, we could condense the proposed language into a clearly presented table ('cuz words are hard :P ), and we could probably use the existing proposed language, too.
- Badges are nice.  We have a 'Partner' badge already, and we could expand this concept over time.  We should also probably have a section in the partners page that describes our badges program so that it's clear to applying partners what they'll get.

## Language
Mongo, gitlab, and debian, do not seem to provide language to the effect of "we'll not hold the project accountable" (if they do, it's not obvious).  
Suse might, but you'd have to go through the partner registration to get the terms and conditions pdf (it might be available elsewhere, but I haven't had a chance to look).

The [partners code of conduct](https://github.com/StackStorm/discussions/issues/51) says a lot (esp. the conflict of interest), but I think it lacks one thing that was discussed during the TSC meeting which is: simple, visible language to say "don't sue ST2, yo".

@armab, the language in the [proposed draft](https://github.com/StackStorm/discussions/issues/73#issuecomment-814308752) also doesn't seem to convey that message (don't sue st2).

I think the key point here is that I, as _not_ a lawyer, don't see anything that explicitly states that ST2 won't be liable.  It's possible it's buried in the verbiage somewhere, but without being well versed in legal-ese, it's not clear, and I suspect that the same will be true for others, so clear, concise language might be helpful?

### Possible Tasks
- Should we just include a simple message (either to the partners page or to the submission request) to say that the applying partner agrees not to hold ST2 accountable?  Again, something to the effect of `I, as a partner, accept to not hold StackStorm accountable [...]`

**Comment by elewzey** [comment]
@raviorch

Thursday, April 15 at 3 PM Eastern would work for me

**Comment by vivekorch** [comment]
Thanks @mickmcgrath13. I will join the meeting with Ravi.

**Comment by mickmcgrath13** [comment]
Invites sent!  Let me know if you did not receive them.

**Comment by arm4b** [comment]
>  Should we just include a simple message (either to the partners page or to the submission request) to say that the applying partner agrees not to hold ST2 accountable? 

Perhaps then we could merge what the LF recommended to us with the proposed "non-accountable" wording:

>    In your [partnership email] proposal, please include the following statement:
>>
>> On behalf of PARTNER NAME ("Participant"), I agree that Participant will comply with the StackStorm Partners Program requirements published at https://stackstorm.com/partners-program/ and Linux Foundation Policies published at https://lfprojects.org/policies/, as updated from time to time, **and agree to not hold StackStorm liable for any damages or loss during or after our participation in the Partners Program.**

**Comment by blag** [comment]
@mickmcgrath13 Can you retitle this issue with the date of the meeting and update the issue description? See #74, #75, and #76 for examples.

I haven't received an invite, but we should be publicly announcing these meetings anyway.

**Comment by mickmcgrath13** [comment]
# 2021-04-15 - Meeting notes

## Discussion about Ravi's proposal

Screenshared the proposal..  Ravi will upload/share

### Liability/Indemnification discussion
Main goal is to remove liability from partners to stackstorm.

Just include: "partner agrees StackStorm liable"
- on the website/landing page?
- in the email they send?
- in the email that we send back to them?

Outcome: in the email they send which will be outlined in the landing page
Mutual indemnification clause will be a case-by-case basis

### Common partner contributions
Types of contributions that can be made.
Goal is for partners to be labeled for their core competencies so that the public can evaluate who to come to (properly communicated on the StackStorm page).

Opportunities for badges?
- engineering, professional services, business development, marketing
- or labels per experience
- or bronze, silver, gold, platnum
    - difficult because of additional mental hops for customers
    - higher level partners would need higher level contributions from StackStorm

Recommendation: Badges based on categories of expertise.

Should add a note that a more structured partner program will come after a certain threshold of partners exist.  Threshold TBD.

### What can stackstorm give back to partners
Really the only way today is to highlight the partners as well as mutual promotion (blogs, webinars, etc).

## Next Steps
- @raviorch to upload/share his proposal so we can iterate on the verbiage
- Iterate and clean up the doc
    - clean up "how and when do partnerships end" ?
- JP or Ravi to get an indemnification clause to include for the email
- Decide on `/partners` or `/partners-program`
- translate into wordpress
- TSC Vote
- Should Existing Partners (Bitovi, Orchestral, etc) also send the email?

# Video Recording
https://drive.google.com/file/d/179qnP_s0Y3AGTLdUIj_-_twds9fx66mp/view

**Comment by mickmcgrath13** [comment]
Partners program draft doc by @raviorch : https://drive.google.com/drive/folders/1x4-9cX-hrvogrSoLc2FE63BH91dUGN9E

**Comment by mickmcgrath13** [comment]
..also, I did record the call.  I'll link it here as soon as I can find where it went :P

**Comment by mickmcgrath13** [comment]
@raviorch fyi: i updated the doc so that anyone can comment but only some can edit

**Comment by mickmcgrath13** [comment]
Here's the recording of the meeting from 2021-04-15:
https://drive.google.com/file/d/179qnP_s0Y3AGTLdUIj_-_twds9fx66mp/view

**Comment by mickmcgrath13** [comment]
There have been several comments/suggestions to the doc:
https://docs.google.com/document/d/1EVD3efr_cJ-amHyHfDBtMZXmLTFpb4-I/edit

I think most could be simply resolved (cc @raviorch ).  I'm happy to do it :)

There are some that will require further discussion, though

**Comment by mickmcgrath13** [comment]
All changes from [the doc](https://docs.google.com/document/d/1EVD3efr_cJ-amHyHfDBtMZXmLTFpb4-I/edit?pli=1) have been updated in [the draft](https://stackstorm.com/stackstorm-partners-program-description/?preview=true)

# Next Steps
- Define labels (table headers or table cells?)
- Submit formal proposal for TSC voting
- Make it live!  (i.e. Page swapping https://stackstorm.com/stackstorm-partners-program-description/?preview=true from https://stackstorm.com/partners-program/)
- Determine the criteria for "How and when do partnerships end? > The support, contribution or service is no longer delivered"
- Revisit partner levels (i.e. bronze, silver, gold) Once a governing committee is defined/established

**Comment by mickmcgrath13** [comment]
> Define labels (table headers or table cells?)

We've generally decided that the labels will be based on the table headers.  I've set up some common symbols to be able to be re-used and added as labels per entry in the partners page:
https://stackstorm.com/partners/

Example:
![image](https://user-images.githubusercontent.com/1030325/120341659-aef71280-c2bc-11eb-86a1-e086d5a62083.png)


Next steps:
- Submit formal proposal for TSC voting
- Make it live! (i.e. Page swapping https://stackstorm.com/stackstorm-partners-program-description/?preview=true from https://stackstorm.com/partners-program/)
- Determine the criteria for "How and when do partnerships end? > The support, contribution or service is no longer delivered"
- Revisit partner levels (i.e. bronze, silver, gold) Once a governing committee is defined/established

**Comment by arm4b** [comment]
Just discussed it in the TSC the screenshot/sketch. This is very helpful to get a visible idea of it, thanks a lot @mickmcgrath13 !

A few comments: 

* Looks like it takes more space on the page extending the entire height. Once we put it together and repeat for each partner, - that might look more overwhelming. More scrolling, fewer partners on a visible page space.
 
* Based on discussions we had, some partners expressed the fact they don't want to be associated with the pre-defined explicit written labeling like that.

* If we'll label PackageCloud, CloudSmith, AWS which provide hosting/infrastructure and associate them with the **Business Development** label, - it's irrelevant and confusing not just for some partners but also users.

* Can we make these labels more compact?
Conceptually, it shouldn't take more visibility than the call-to-action **More Info** button, which is the main URL and landing page target for each partner. Normally Partners would be interested to make the most visible call-to-action URL button and not the label.

* How about positioning labels just as icons near the partner's name?
  > **Bitovi**  *label-icon1* *label-icon2*

* We'll add a tooltip for each icon (when the user places his cursor mouse over the icon) hinting if it's a Business Development, Professional Services, Marketing, etc.
This way partners contributing to more categories will just have more icon labels highlighting their level of involvement. It won't overwhelm design much, serve the goal and won't repel potential partners nor confuse users.

See:
![st2-labels-sketch_refined](https://user-images.githubusercontent.com/1533818/120384522-3032c180-c31e-11eb-99b1-0a8998ff33cd.png)

**Comment by mickmcgrath13** [comment]
@armab I agree the large labels are a bit much.  Icons with tooltips are doable and gives the best of both worlds.

Placing icons beside the partner name is tricky with the thrive architect, so below the name would be easier.  We can do icons below easily with a tooltip:
![image](https://user-images.githubusercontent.com/1030325/120399439-a4ab3780-c301-11eb-98a4-275beba27394.png)

with tooltip:
![image](https://user-images.githubusercontent.com/1030325/120399454-ac6adc00-c301-11eb-8a80-f866ddb9f316.png)



Also, I'm not sure I understand this:
> Based on discussions we had, some partners expressed the fact they don't want to be associated with the pre-defined explicit written labeling like that.

and this:
> If we'll label PackageCloud, CloudSmith, AWS which provide hosting/infrastructure and associate them with the Business Development label, - it's irrelevant and confusing not just for some partners but also users.


are we going with the 4 icons which are currently the headers of the table in the draft page?  if not, what all labels/icons do we need?  Currently, we have:
![image](https://user-images.githubusercontent.com/1030325/120399667-15525400-c302-11eb-8490-e5800f0aa797.png)


If this is not what we want for the labels, please recommend a list of labels (and associated icons).  IMO, these 4 'categories' should be sufficient until such time as a partner says "i'd like something different"

**Comment by mickmcgrath13** [comment]
also, fyi, to add the labels, do the following:
- insert a content box under the partner name text box
    - remove all margin and padding except for margin-bottom of 20
- insert icon into content box
    - remove top and bottom margin
    - left and right margin of 7
    - set padding of 7 all around
    - set size to 20
    - update colors (orange BG, white icon)
    - set alignment to the left, and float
    - change "state" to "hover"
    - add tooltip with the name of the label (position top)
    - repeat for other icons

**Comment by arm4b** [comment]
![](https://user-images.githubusercontent.com/1030325/120399454-ac6adc00-c301-11eb-8a80-f866ddb9f316.png)

:100: That would work! :+1:  @mickmcgrath13 

Or even switching from Thrive Architect to raw HTML for that page is OK too to make the icons inline. 

After all, we used the tool to get this page initially working and with a stable Partners version we won't need to implement anything new. Just edit old partners and add new ones.

Eventually, it'll be static HTML anyways as we migrate out from the WordPress.

**Comment by mickmcgrath13** [comment]
I went down the plain HTML route before, and it didn't seem quite as straight-forward as I had hoped.

I found a way to get the icons beside the partner with Thrive, anyway:
- add content box
    - remove margins and padding
- add text box with partner name
    -  align left
    - display inline
    - float
    - margin-right: 13 (remove all other margins and padding)
- add icon
    -  align left
    - display inline
    - float
    - margin-right: 7 (remove all other margins and padding)
    - set size to 18
    - set padding to 4 all around
    - update colors (orange BG, white icon)
    - change "state" to "hover"
    - add tooltip with the name of the label (position top)
    - repeat for other icons

It's also helpful to make use of the copy functionality as it copies all the preset settings from the module that is copied.

Here's an example:
![image](https://user-images.githubusercontent.com/1030325/120479706-df4db800-c373-11eb-9f39-d2a1727ac111.png)
![image](https://user-images.githubusercontent.com/1030325/120479737-e8d72000-c373-11eb-84b8-e11ff4572990.png)

**Comment by mickmcgrath13** [comment]
Official Vote here: https://github.com/StackStorm/discussions/issues/80

**Comment by mickmcgrath13** [comment]
Closing as all the pages are live!
